pkg/server/singleprocess/auth.go

```diff
@@ -237,6 +237,9 @@ func (s *Service) authRunner(
 		(r.AdoptionState != pb.Runner_ADOPTED &&
 			r.AdoptionState != pb.Runner_PREADOPTED)
 	if notAdopted {
+		// We sleep here to tarpit any runaway runners that are going to thrashing
+		// trying to connect over and over and over even though they're not allowed in.
+		time.Sleep(5 * time.Second)
```

Comment:
I know this is a temp fix but would we want some kind of rolling backoff for the time we sleep/retry over time?

Reply:
there isn't really the ability to do that, because we don't have a stable criteria to use to change the sleep time.